tests/install_testbed/launch_vm.py

```diff
@@ -0,0 +1,192 @@
+#
+# You may obtain a copy of the License at:
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# For those usages not covered by the Apache version 2.0 License please
+# contact with opensource@tid.es
+#
+__author__ = 'chema'
+
+import time
+import sys
+import os
+import os.path
+
+from utils.osclients import osclients
+import settings
+
+
+def launch_vm(vm_n, flavor_n, securityg_n, image_n, ifaces, user_data=None):
+    """ Launch a VM and wait until it is ready.
+
+    :param vm_n: virtual machine name
+    :param flavor_n: flavor name
+    :param securityg_n: securigy group name
+    :param image_n: image name
+    :param ifaces: array with the interfaces
+    :param user_data: optional init script or cloud init data
+    :return: a VM object
+    """
+    # Get image
+    image_id = osclients.get_glanceclient().images.find(name=image_n).id
+
+    nova_c = osclients.get_novaclient()
+    flavor = nova_c.flavors.find(name=flavor_n)
+
+    extra_params = dict()
+    
+    # load script to launch with nova.servers.create()
+    if user_data:
+        data = open(user_data).read()
+        extra_params['userdata'] = data
+
+    server = nova_c.servers.create(
+        vm_n, flavor=flavor.id, image=image_id, key_name=settings.key_name,
+        security_groups=[securityg_n], nics=ifaces, **extra_params)
+
+    print('Created VM with UUID ' + server.id)
+
+    # wait until the vm is active
+    tries = 1
+    while server.status != 'ACTIVE' and tries < 30:
+        print('Waiting for ACTIVE status. (Try ' + str(tries) + '/30)')
+        time.sleep(5)
+        server = nova_c.servers.get(server.id)
+        tries += 1
+
+    if server.status != 'ACTIVE':
+        sys.stderr.write('Failed waiting the VM is active\n')
+        sys.exit(-1)
+
+    return server
+
+def create_port_multi_ip(security_group_id=None):
+    """Create port in external network, with 5 IPs (this is the maximum allowed
+    with the default neutron configuration).
+
+    Be aware that this port is not deleted when the VM is deleted!
+
+    Another point is that security group may be different than the security
+    group of the VM.
+
+    :param security_group_id: The security group
+    :return: a pre-allocated port
+    """
+
+    subnet_external = None
+    for subnet in neutron.list_subnets()['subnets']:
+        if subnet['network_id'] == network['external']:
+            subnet_external = subnet['id']
+
+    fixed_ips = list()
+    for i in range(200, 205):
+        fixed_ip = dict()
+        fixed_ip['subnet_id'] = subnet_external
+        fixed_ip['ip_address'] = '192.168.58.%d' % i
```

Comment:
Yes, it is a magic number to use this network, but the problem is that moving this to a python configuration file, even with the simplification of only allowing a fixed network mask, also implies other serious changes like parsing the configuration from a shell scripts (the create_resources.sh) and provide more and more  explanations in the README about what is the floating IP that is obtained, when nothing of this is useful to provide a testbed to test skuld. 

Indeed, all this code is not used at this moment, because only multinetwork == False is supported. When current configuration we are using some specific networks also in prepare_nodes.sh